test/Confluent.Kafka.IntegrationTests/Tests/AssignPastEnd.cs

```diff
@@ -63,8 +68,19 @@ public static void AssignPastEnd(string bootstrapServers, string topic)
                     Assert.True(false);
                 };
                 consumer.Poll(TimeSpan.FromSeconds(10));
+                consumer.Assign(new List<TopicPartitionOffset>() { new TopicPartitionOffset(dr.TopicPartition, dr.Offset+2) });
+                consumer.Poll(TimeSpan.FromSeconds(10));
             }
 
+            consumerConfig["default.topic.config"] = new Dictionary<string, object>() { { "auto.offset.reset", "earliest" } };
+            using (var consumer = new Consumer(consumerConfig))
+            {
+                Message msg;
+                consumer.Assign(new List<TopicPartitionOffset>() { new TopicPartitionOffset(dr.TopicPartition, dr.Offset+1) });
```

Comment:
Given there was some confusion over what the behavior should be depending on lastOffset+1 vs lastOffset+2, you might want to add some comments explaining the expected behavior (inline w/ the asserts or the to comment on the test.